packages/components/src/Autocomplete/Autocomplete.types.ts

```diff
@@ -0,0 +1,140 @@
+import { RefObject } from "react";
+import { FormFieldProps } from "../FormField";
+
+type OptionValue = string | number;
+
+export interface BaseOption {
+  label: string;
+}
+
+export interface Option extends BaseOption {
+  value?: OptionValue;
+  description?: string;
+  details?: string;
+}
+
+export interface GroupOption extends BaseOption {
+  options: Option[];
+}
+
+export type OptionCollection = AnyOption[];
+export type AnyOption<
+  GenericOption extends Option | GroupOption = Option | GroupOption,
+> = GenericOption;
+
+export interface AutocompleteProps<
+  GenericOption extends AnyOption = AnyOption,
+  GenericOptionValue extends Option = Option,
+  GenericGetOptionsValue extends AnyOption = AnyOption,
```

Comment:
The generic on GetOptionsValue is because some people were passing an initialOptions value that would be a different type to what was returned by `getOptions` e.g. Initial options would be a list of GroupOptions but getOptions would give Options based on another value in the form